### ISSUE TITLE
Gradle: Vaadin doesn't reuse production bundle from vaadin-prod-bundle.jar

### ISSUE THREAD

**Issue by mvysny** [vaadin/flow#18010]
### Description of the bug

When using the following simple `build.gradle.kts` file, Vaadin plugin always builds the production bundle and never reuses the pre-built production bundle from `vaadin-prod-bundle.jar` since it's missing on the classpath:

```
Passing this classpath to NodeTasks.Builder: [vaadin-core-24.2.2.jar, vaadin-core-internal-24.2.2.jar, vaadin-dev-24.2.2.jar, flow-lit-template-24.2.2.jar, flow-html-components-24.2.2.jar, flow-data-24.2.2.jar, flow-dnd-24.2.2.jar, vaadin-confirm-dialog-flow-24.2.2.jar, vaadin-dev-bundle-24.2.2.jar, vaadin-dev-server-24.2.2.jar, flow-server-24.2.2.jar, flow-push-24.2.2.jar, flow-client-24.2.2.jar, vaadin-lumo-theme-24.2.2.jar, vaadin-material-theme-24.2.2.jar, vaadin-accordion-flow-24.2.2.jar, vaadin-avatar-flow-24.2.2.jar, vaadin-upload-flow-24.2.2.jar, vaadin-app-layout-flow-24.2.2.jar, vaadin-button-flow-24.2.2.jar, vaadin-checkbox-flow-24.2.2.jar, vaadin-combo-box-flow-24.2.2.jar, vaadin-custom-field-flow-24.2.2.jar, vaadin-date-time-picker-flow-24.2.2.jar, vaadin-date-picker-flow-24.2.2.jar, vaadin-details-flow-24.2.2.jar, vaadin-time-picker-flow-24.2.2.jar, vaadin-select-flow-24.2.2.jar, vaadin-side-nav-flow-24.2.2.jar, vaadin-dialog-flow-24.2.2.jar, vaadin-form-layout-flow-24.2.2.jar, vaadin-field-highlighter-flow-24.2.2.jar, vaadin-grid-flow-24.2.2.jar, vaadin-icons-flow-24.2.2.jar, vaadin-virtual-list-flow-24.2.2.jar, vaadin-list-box-flow-24.2.2.jar, vaadin-login-flow-24.2.2.jar, vaadin-messages-flow-24.2.2.jar, vaadin-ordered-layout-flow-24.2.2.jar, vaadin-progress-bar-flow-24.2.2.jar, vaadin-radio-button-flow-24.2.2.jar, vaadin-notification-flow-24.2.2.jar, vaadin-menu-bar-flow-24.2.2.jar, vaadin-context-menu-flow-24.2.2.jar, vaadin-renderer-flow-24.2.2.jar, vaadin-split-layout-flow-24.2.2.jar, vaadin-tabs-flow-24.2.2.jar, vaadin-text-field-flow-24.2.2.jar, throw-if-servlet3-1.0.2.jar, ph-css-7.0.1.jar, atmosphere-runtime-3.0.3.slf4jvaadin1.jar, license-checker-1.12.3.jar, directory-watcher-0.18.0.jar, ph-commons-11.1.0.jar, oshi-core-6.4.1.jar, slf4j-api-2.0.9.jar, gwt-elemental-2.8.2.vaadin2.jar, commons-fileupload2-jakarta-2.0.0-M1.jar, open-8.5.0.jar, commons-fileupload2-core-2.0.0-M1.jar, commons-io-2.13.0.jar, jackson-annotations-2.15.3.jar, jackson-datatype-jsr310-2.15.3.jar, jackson-databind-2.15.3.jar, jackson-core-2.15.3.jar, jsoup-1.16.1.jar, asm-9.5.jar, gentyref-1.2.0.vaadin1.jar, commons-compress-1.24.0.jar, vaadin-flow-components-base-24.2.2.jar, commons-lang3-3.13.0.jar, javaparser-core-3.25.5.jar, nimbus-jose-jwt-9.31.jar, jcip-annotations-1.0.0.jar, jsr305-3.0.2.jar, jna-platform-5.13.0.jar, jna-5.13.0.jar, /home/mavi/work/my/myproject/build/resources/main]
Scanning classes to find frontend configurations and dependencies...
Visited 618 classes. Took 350 ms.
Checking if a production mode bundle build is needed
There is no production bundle in the classpath.
No bundle's stats.json found for production-bundle validation.
A production mode bundle build is needed
Running `npm install` to resolve and optionally download frontend dependencies. This may take a moment, please stand by...
```

### Expected behavior

The `vaadin-prod-bundle.jar` should be on the classpath and Vaadin plugin should use it.

### Minimal reproducible example

Create this file named `build.gradle.kts`:
```kotlin
plugins {
    java
    id("com.vaadin") version "24.2.2"
}

repositories {
    mavenCentral()
}

dependencies {
    // Vaadin
    implementation("com.vaadin:vaadin-core:24.2.2")
}
```
There are no other files needed - no java routes or anything. Now run `./gradlew clean build -Pvaadin.productionMode --info`.

### Versions

- Vaadin / Flow version: 24.2.2
- Java version: 17
- OS version: ubuntu

**Comment by mvysny** [comment]
The vaadin-prod-bundle.jar is not added to flow-gradle-plugin, but only later in vaadin-gradle-plugin, in this template: https://github.com/vaadin/platform/blob/4b531899e88621157dbd918f870818abe889f809/scripts/generator/templates/template-vaadin-gradle-plugin-pom.xml#L43

However, I think that Gradle generates pom.xml from instructions stored in build.gradle, thus overwriting the template :thinking_face: Therefore it would probably be better to include the dependency on prod-bundle into plugin's build.gradle script.

It could be that the template is overwritten, BUT the prod-bundle is included in some different way in some projects (e.g. if you use BOM)?

**Comment by mvysny** [comment]
I don't quite understand the layout of Gradle plugin repository, but if you take a look at https://plugins.gradle.org/m2/com/vaadin/ then you'll find:

1. https://plugins.gradle.org/m2/com/vaadin/vaadin-gradle-plugin/24.2.2/vaadin-gradle-plugin-24.2.2.pom doesn't refer to the vaadin-prod-bundle dependency.
2. Also https://plugins.gradle.org/m2/com/vaadin/com.vaadin.gradle.plugin/24.2.2/com.vaadin.gradle.plugin-24.2.2.pom doesn't refer to the vaadin-prod-bundle dependency.

**Comment by mvysny** [comment]
Adding this to the build script fixes the issue for some reason:
```groovy
buildscript {
    repositories {
        mavenCentral()
    }
}
```

**Comment by mvysny** [comment]
I've figured it out. Adding the `mavenCentral()` buildscript repository causes Gradle to prioritize Maven Central over plugins.gradle.org. And indeed, the pom.xml for the plugin looks very different on Maven Central: https://repo1.maven.org/maven2/com/vaadin/vaadin-gradle-plugin/24.2.2/vaadin-gradle-plugin-24.2.2.pom

It now contains a reference to vaadin-prod-bundle.jar, which then causes Gradle to add prod-bundle.jar on classpath.

This is indeed very surprising, having two different pom.xmls in two official repositories; I don't think this is a good practice.

**Comment by mvysny** [comment]
Awesome, thank you so much for such a quick fix!

**Comment by mvysny** [comment]
I've verified that the fix is also present in Vaadin 24.2.3 :partying_face:
